transformation/plugins/org.jboss.tools.fuse.transformation.editor/src/main/java/org/jboss/tools/fuse/transformation/editor/wizards/NewTransformationWizard.java

```diff
@@ -434,14 +432,16 @@ private String generateModel(String filePath,
             return getJavaQualifiedName(filePath, monitor);
         }
         case JSON: {
+            String className = classNameBuilder.toString().replace("_", "");
```

Comment:
Rethinking about it after your comment. I think that JsonModelGeenrator should be responsible to manage the special characters and in this case it will be easier to test the JSonModelGenerator